File: java/ankit_kanyal/devnest_problems/phase_1/ReverseLinkedList.java
// link- https://devsnest.in/algo-challenges/reverse-linked-list

static Node solve(Node head){
//CODE HERE 
    Node prev=null;
	Node cur=head;
	Node next=null;
		
	while(cur!=null){
		next=cur.next;
        cur.next=prev;
	    prev=cur;
	    cur=next;
			
	}
	return prev;
}

File: python/vedant_bhatnagar/DevsnestProblems_THA/Phase 1/reverseLinkedList.py
def solve(head):
    # CODE HERE
    prev_node = next_node = None
    curr = head
    while curr != None:
        next_node = curr.next
        curr.next = prev_node
        prev_node = curr
        curr = next_node
    head = prev_node
    return head


    #second approach
    if not head or not head.next:
        return head
    p,c,n = head, head.next, head.next.next
    while n:
        c.next = p
        p = c
        c = n
        n = n.next
    head.next = None
    c.next = p
    return c

    #recursion
    if not head:
        return None
    if head.next == None:
        return head
    reverse = solve(head.next)
    head.next.next = head
    head.next = None
    return reverse